### PR TITLE
Add modes for the emission profile bar plot without financial data

### LINKED ISSUE #133: Add modes for bar plot without financial data

### PR DESCRIPTION
**Pull request by lindadelacombaz**
Closes #133.
Dear @maurolepore,
I have enhanced the bar_plot_emission_profile() function, adding the different modes as arguments. 
The PR ended up being also a refactoring one for the bar_plot_emission_profile() function.

----

TODO

- [x] [Link related issue/PR]([url](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)). 
- [x] Describe the goal of the PR. Avoid details that are clear in the diff.
- [x] Mark the PR as draft.
- [x] [Include a unit test](https://code-review.tidyverse.org/reviewer/aspects.html#sec-tests).
- [x] Review your own PR in "Files changed".
- [x] Ensure the PR branch is updated.
- [x] Ensure the checks pass.
- [x] Change the status from draft to ready.
- [x] Polish the PR description to reflect what it ended up doing.
- [x] Polish the PR title as you'd like others to read it from the git log.
- [x] Assign a reviewer.
- [x] Document user-facing changes in the changelog.

### CODE REVIEW

File: R/bar_plot_emission_profile.R [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #' @param benchmarks A character vector specifying the benchmarks for which the
 #' emission profiles will be plotted. The user can choose from one to several
 #' benchmark(s) to be plotted.
+#' @param mode A character vector.
```
[the remark this answers is not in the document]
Comment (maurolepore) [reply R/bar_plot_emission_profile.R:10]: This way you can be more specific and at the same time stay flexible.

File: R/bar_plot_emission_profile.R [hunk 2]
```diff
@@ -16,44 +17,70 @@
...
+#' @param data A data frame.
+#'
+#' @return A data frame
+#' @noRd
+check_bar_plot_emission_profile <- function(data) {
   crucial <- c(
     "benchmark",
+    "equal_weight",
+    "best_case",
+    "worst_case",
```
Comment (maurolepore) [inline R/bar_plot_emission_profile.R:43]: This can use `modes()`

File: tests/testthat/test-bar_plot_emission_profile.R [hunk 3]
```diff
@@ -1,30 +1,86 @@
...
+})
+
+test_that("proportions are less or equal to 1 for worst case mode", {
+  data <- example_without_financial()
+  data <- data |>
+    prepare_bar_plot_emission_profile(benchmarks(), mode = "worst_case")
+
+  proportions <- data$proportion
+  expect_true(all(proportions >= 0 & proportions <= 1))
 })
```
Comment (maurolepore) [inline tests/testthat/test-bar_plot_emission_profile.R:86]: Nice! The `prepare*()` functions now belong to a different module and is easier to test than the visuals. That module can stay here or if you want to enforce stronger boundaries then we can move it to a separate package, e.g. tilltPlotPrepare.

**Review by maurolepore — APPROVED** [review]
LGTM. Thanks Linda.

Note I suggested some changes that you may commit right on the PR

image


[Video](https://www.linkedin.com/posts/mauro-lepore_suggest-changes-directly-on-a-github-pr-activity-7203730894824587264-ahOm?utm_source=share&utm_medium=member_desktop)

File: DESCRIPTION [hunk 4]
```diff
@@ -32,14 +32,16 @@ Imports:
     tidyr,
     tidyselect,
     tiltIndicator,
+    tiltIndicatorAfter (>= 0.0.0.9042),
     tools,
     vroom
 Suggests: 
     testthat (>= 3.0.0),
```
[the remark this answers is not in the document]
Comment (lindadelacombaz) [reply DESCRIPTION:38]: @maurolepore It makes the R CMD check fails, I kept it in the Imports: section

**Pull request merged**